### DISCUSSION TITLE
Matter servers and devices

### DISCUSSION THREAD

**Discussion by Nik19792000** [FezVrasta/node-red-matter#16] [General]
Hi, I have some doubts about the structure of node-red-matter nodes:
1. Matter Server node: what's the benefit of having multiple servers defined? It allows you to reuse the same ports, but that will naturally clash. Why not define a singleton in the code, which manages the Matter Server instance in matter.js?
2. Device Control/Device Status: why are there two UI nodes for control/status that link to device node instead of directly defining the device node as UI node? It can have an input for control and an output for status. The current way seems to be unnecessarily complex.

**Comment by FezVrasta** [comment]
Having multiple servers can be useful for advanced use cases.

Having a way to define multiple input/output nodes nodes for the same device means you can have multiple flows that reference the same device
